### ISSUE TITLE
Suggest changing `if let Some(3) = x` to `if x == Some(3)`

### ISSUE THREAD

**Issue by adamchalmers** [rust-lang/rust-clippy#7554]
### What it does

I was mentoring a new Rust programmer on my team, and he was a bit confused by `if let`. The root of his confusion was that he thought `if let` is used to introduce a binding. I agreed with him! But then he showed me a case where you can use `if let` without binding:

```rust
if let Some(3) = x {
   println!("it's true);
}
```

I told him, "yeah, if I saw that in code review, I would suggest using `if x == Some(3)` instead.

### Categories (optional)

- Kind: style

*What is the advantage of the recommended code over the original code*

I think it's simpler, it uses a more standard language construct. You're just evaluating a boolean condition. IMO you should only use `if let` if you're actually trying to pattern match, but simply checking equality doesn't require pattern matching!

### Drawbacks

This is basically just a generalization of [redundant_pattern_matching](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching), so it might have the same problems with ordering which that lint has.

### Example

```rust
if let Some(3) = x {
   println!("it's true);
}
```

Could be written as:

```rust
if x == Some(3) {
   println!("it's true);
}
```

**Comment by camsteffen** [comment]
Duplicate of #1716
